BanTheBad/middleware
# chore(folder-structure): Update to be modular

#### What does this PR do?
Update the codebase to be modular and to some extent follow the domain driven design pattern

- infra/
  - sequelize-orm/
  - swagger/
- src/
  - accused/
  - admin/
  - auth/
  - cases/
  - shared/
  - users/
  - victims/

#### Description of task to be completed?
So, we should have these modules and code changes related to these modules should be done in them to help in better decoupling the codebase, maintaining individual modules separately without affecting other modules and maybe packaging each modules if needed as a package to be used in another codebase.

#### How should this manually be tested?
- Follow the `README.md` to setup the application
- Ensure the `.env` file created is filled with correct credentials for your database etc.
- Follow instructions here #15 to run the migrations
- Do `npm run dev` in the terminal to start in development
- Visit `localhost:5000/users` and when you reload the page, a dummy user should be created in the database and shown in the browser

#### What are the relevant user stories?
N/A

#### Screenshots (if appropriate)
<img width="987" alt="Screenshot 2020-10-26 at 21 14 21" src="https://user-images.githubusercontent.com/20375577/97226666-9974b400-17d4-11eb-90e8-4e62045969cf.png">

<img width="1138" alt="Screenshot 2020-10-26 at 21 46 03" src="https://user-images.githubusercontent.com/20375577/97226713-aabdc080-17d4-11eb-9402-dfe197139aff.png">

<img width="363" alt="Screenshot 2020-10-26 at 21 47 51" src="https://user-images.githubusercontent.com/20375577/97226865-e9ec1180-17d4-11eb-98ae-5099626c19f9.png">
